plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/PageValidator.java

```diff
@@ -31,11 +31,17 @@ public PageValidator(AppEngineWizardPage wizardPage) {
 
   @Override
   public void modifyText(ModifyEvent event) {
-    wizardPage.setPageComplete(wizardPage.validatePage());
+    run();
```

Comment:
`PageValidator` was originally intended to be suitable to be added directly as an SWT event listener.  Much of it would be moot with Java 8 lambdas.

TBH, I think this kind of class obfuscates more than it clarifies.  I'll rewrite it to just use a `revalidate()` method.